src/components/dialogs/dataSources/COVIDcast.svelte

```diff
@@ -23,34 +24,41 @@
     }
   }
 
-  onMount(() => {
-    fetchCOVIDcastMeta().then((res) => {
-      geoTypes = [...new Set(res.map((d) => d.geo_type))];
-      const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
-      for (const row of res) {
-        const ds = byDataSource.get(row.data_source);
-        if (!ds) {
-          byDataSource.set(row.data_source, {
-            label: row.data_source,
-            value: row.data_source,
-            signals: [row.signal],
-          });
-        } else if (!ds.signals.includes(row.signal)) {
-          ds.signals.push(row.signal);
+  function fetchMetadata() {
+    fetchCOVIDcastMeta((api_key = api_key)).then((res) => {
+      if (res.length !== 0) {
+        geoTypes = [...new Set(res.map((d) => d.geo_type))];
+        const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
+        for (const row of res) {
+          const ds = byDataSource.get(row.data_source);
+          if (!ds) {
+            byDataSource.set(row.data_source, {
+              label: row.data_source,
+              value: row.data_source,
+              signals: [row.signal],
+            });
+          } else if (!ds.signals.includes(row.signal)) {
+            ds.signals.push(row.signal);
+          }
         }
+        byDataSource.forEach((entry) => {
+          entry.signals.sort();
+        });
+        dataSources = [...byDataSource.values()].sort((a, b) => a.value.localeCompare(b.value));
       }
-      byDataSource.forEach((entry) => {
-        entry.signals.sort();
-      });
-      dataSources = [...byDataSource.values()].sort((a, b) => a.value.localeCompare(b.value));
     });
+  }
+
+  onMount(() => {
+    fetchMetadata();
   });
 
   export function importDataSet() {
-    return importCOVIDcast({ data_source, signal, geo_type, geo_value });
+    return importCOVIDcast({ data_source, signal, geo_type, geo_value, api_key });
   }
 </script>
 
+<TextField id="{id}-api" label="API Key" name="api_key" on_change={() => fetchMetadata()} bind:value={api_key} />
```

Comment:
This should not be a required field:
```suggestion
<TextField id="{id}-api" label="API Key (optional)" name="api_key" on_change={() => fetchMetadata()} bind:value={api_key} />
```
i see this when trying to submit with an empty input:
![image](https://github.com/user-attachments/assets/2cdf3ef9-4047-4d07-81a5-87999ac5a838)


Can you also make just the "API Key" part of the label clickable, opening this page in a new window? https://cmu-delphi.github.io/delphi-epidata/api/api_keys.html